cpp/src/arrow/compute/kernels/boolean.cc

```diff
@@ -0,0 +1,156 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/compute/kernels/boolean.h"
+
+#include "arrow/compute/context.h"
+#include "arrow/compute/kernels/util-internal.h"
+#include "arrow/util/logging.h"
+
+#include <vector>
+
+namespace arrow {
+namespace compute {
+
+class InvertKernel : public UnaryKernel {
+  Status Call(FunctionContext* ctx, const Datum& input, Datum* out) override {
+    DCHECK_EQ(Datum::ARRAY, input.kind());
+
+    const ArrayData& in_data = *input.array();
+    ArrayData* result;
+
+    if (out->kind() == Datum::NONE) {
```

Comment:
At first sight, it doesn't seem obvious what this is about. Can you comment on this?